tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/IHttpControllerContext.cs

```diff
@@ -14,6 +14,8 @@ internal interface IHttpControllerContext
         IHttpRequestMessage Request { get; }
 
         IHttpRouteData RouteData { get; }
+
+        RequestContextStruct RequestContext { get; }
```

Comment:
Since the rest of the members are interfaces, maybe this should also be an interface? 🤷🏼 I don't really know to be honest

Reply:
I think it should be a struct [based on the best practice graph](https://github.com/DataDog/dd-trace-dotnet/blob/master/docs/development/DuckTyping.md#best-practices), 

> Caveat, I added that graph (but based on @tonyredondo's testing and advice!) 

The `IHttpControllerContext` makes sense as an interface because it's used in a constraint in the integration directly, but I don't think `IHttpRequestMessage` or `IHttpRouteData` should be, I think they should be duckcopy structs. I haven't changed them just in case I'm wrong, but I'm pretty sure we could 😄